### PR TITLE
Progress test

### PR DESCRIPTION
**Pull request by ikerBorr**
### Description
This pull request includes updates to the latest solutions for the breakpoint test and fixes for some additional bugs. However, the following items remain pending:

### Pending Issues
1. **Tests to be resolved:**
   - `rv32mi-p-shamt`
   - `rv32uc-v-rvc`
   - `rv32uzfh-p-fcvt_w`
   - `rv32uzfh-v-fcvt_w`

2. **Areas to be reviewed:**
   - Cycle count and the method for detecting the end of the test in virtual memory tests.
   - Implementation of the `clmul` instruction.
   - Handling of specific memory access exceptions when checking PTE flags.

### CODE REVIEW

File: punxa/fpu.py [hunk 1]
```diff
@@ -1273,7 +1273,7 @@ def convert_hp_to_i32(self, v):
             if (a.compare(b) != 0):
                 self.cpu.csr[CSR_FFLAGS] = CSR_FFLAGS_INEXACT_MASK
                 
-        return signExtend(r, 32, 64) 
+        return r#signExtend(r, 32, 64)
```
Comment (davidcastells) [inline punxa/fpu.py:1276]: fpu is shared with 64 bit version of the processor, are you shure your change to convert_hp_to_i64 is not breaking the 64 bits version ?
Why RV32 processor should be using convert_hp_to_i64 ?

**Review by davidcastells — CHANGES_REQUESTED** [review]
can you check that your changes to fpu do not break RV64 version ?

File: punxa/fpu.py (hunk 1 above)
Comment (ikerBorr) [reply punxa/fpu.py]: Resolved. There was an incompatibility with the functions: convert_sp_to_i32, convert_hp_to_u32, and convert_hp_to_i32. I have duplicated the functions to create separate versions for 64-bit and 32-bit systems.

**Review by davidcastells — APPROVED** [review]
not very neat , but I'll fix it

**Pull request merged**